lib/boxr/auth.rb

```diff
@@ -53,6 +55,33 @@ def self.revoke_tokens(token, client_id: ENV['BOX_CLIENT_ID'], client_secret: EN
     auth_post(uri, body)
   end
 
+  # Exchange a fully-scoped token for one that is down-scoped to specific permissions,
+  # defined by the scopes passed in.  You can optionally restrict the token to a
+  # particular file.
+  #
+  # @param [String] subject_token The fully-scoped token to exchange
+  #
+  # @param [Array<String>] scopes A list of the scopes to allow in the new token
+  #
+  # @param [Integer, String] file (Optional) The ID of a file resource to restrict
+  #   the token to
+  # @return [BoxrMash]
+  def self.exchange_token(subject_token, scopes:, file: nil)
```

Comment:
Maybe specify that it's `file_id`. The current codebase doesn't do that much but I think it's a slight improvement for clarity